src/targets/awsLambda.ts

```diff
@@ -0,0 +1,187 @@
+   * @param _version New version to be released.
+   * @param revision Git commit SHA to be published.
+   */
+  public async publish(_version: string, revision: string): Promise<any> {
+    logger.debug('Fetching artifact list...');
+    const packageFiles = await this.getArtifactsForRevision(revision, {
+      includeNames: DEFAULT_AWS_LAMBDA_DIST_REGEX,
+    });
+
+    if (packageFiles.length == 0) {
+      reportError('Cannot release to AWS Lambda: no packages found');
+      return undefined;
+    } else if (packageFiles.length > 1) {
+      reportError(`Cannot release to AWS Lambda:
+      multiple packages with matching patterns were found.`);
+      return undefined;
+    }
+
+    const artifactBuffer = fs.readFileSync(
+      await this.artifactProvider.downloadArtifact(packageFiles[0])
+    );
+
+    const publishRegionToLambda = async (currentRegion: string) => {
+      const lambda = new Lambda({ region: currentRegion });
+
+      const publishedLayer = await this.publishAwsLayer(lambda, {
+        Content: {
+          ZipFile: artifactBuffer,
+        },
+        LayerName: AwsLambdaTarget.layerName,
+        CompatibleRuntimes: compatibleRuntimes,
+        LicenseInfo: 'MIT',
+      });
+
+      if (publishedLayer.Version === undefined) {
+        reportError(`Error while publishing AWS Layer to ${currentRegion}`);
+        return;
+      }
+
+      await this.addAwsLayerPermissions(lambda, {
+        LayerName: AwsLambdaTarget.layerName,
+        VersionNumber: publishedLayer.Version,
+        StatementId: 'public',
+        Action: 'lambda:GetLayerVersion',
+        Principal: '*',
+      });
+
+      logger.info(`Published layer in ${currentRegion}:
+        ${publishedLayer.LayerVersionArn}`);
+    };
+
+    await Promise.all(awsAllRegions.map(publishRegionToLambda));
+  }
+
+  /**
+   * Publishes the layer to AWS Lambda with the given layer data.
+   * It must contain the buffer for the ZIP archive and the layer name.
+   * Each time you publish with the same layer name, a new version is created.
+   * @param lambda The lambda service object.
+   * @param layerData Details of the layer to be created.
+   */
+  public async publishAwsLayer(
+    lambda: Lambda,
+    layerData: Lambda.PublishLayerVersionRequest
+  ): Promise<PromiseResult<Lambda.PublishLayerVersionResponse, AWSError>> {
+    return await lambda.publishLayerVersion(layerData).promise();
+  }
+
+  /**
+   * Adds to a layer usage permissions to other accounts.
+   * @param lambda The lambda service object.
+   * @param layerPermissionData Details of the layer and permissions to be set.
+   */
+  public async addAwsLayerPermissions(
+    lambda: Lambda,
+    layerPermissionData: Lambda.AddLayerVersionPermissionRequest
+  ): Promise<
+    PromiseResult<Lambda.AddLayerVersionPermissionResponse, AWSError>
+  > {
+    return await lambda
+      .addLayerVersionPermission(layerPermissionData)
+      .promise();
+  }
+}
```

Comment:
Why wrap these one-liners into methods? Wouldn't it be the same to mock the Lambda client in tests and write less code?